File: Stacks and Queues/Stacks/Tower_of_Hanoi.c
#include<stdio.h>
int i=0;
void TowerofHanoi(int n, char from, char to, char help){
    // printf("\n");
    if (n==1){
        i++;
        printf("Move Disk 1 from rod %c to rod %c\n", from, to);
        return;
    }

    TowerofHanoi(n-1, from, help, to);
    printf("Move Disk %d from rod %c to rod %c\n", n, from, to);
    i++;
    TowerofHanoi(n-1, help, to, from);
}


int main(){
    int n;
    printf("Enter no of rings : ");
    scanf("%d", &n);
    TowerofHanoi(n, 'A', 'C', 'B');
    printf("Moves = %d\n", i);
}

File: Stacks and Queues/Stacks/TowerofHanoi.py
def Tower(n, frm, to, help):
    if n==1:
        ar[0]+=1
        print("Move Disk 1 from {} to {}".format(frm, to))
        return
    
    Tower(n-1, frm, help, to)
    print("Move Disk {} from {} to {}".format(n, frm, to))
    ar[0]+=1
    Tower(n-1, help, to, frm)


ar=[0]
n = int(input("Enter no of disks : "))
Tower(n, 'A', 'C', 'B')
print("Moves = ",ar[0])